app/javascript/chat/connectReducer.js

```diff
@@ -0,0 +1,320 @@
+import * as Type from './components/ChatTypes';
+export function connectReducer(state, action) {
+  const { type, payload = {} } = action;
+
+  switch (type) {
```

Comment:
refactor (non-blocking): consider combining reducers instead of a super long switch case.

Reply:
I will do these changes incoming PRs with more refactoring.